x-pack/plugins/cases/public/containers/use_delete_comment.tsx

```diff
@@ -5,106 +5,40 @@
 import * as i18n from './translations';
 
-interface CommentDeleteState {
-  isError: boolean;
-}
-interface CommentDelete {
-  commentId: string;
-}
-
-type Action =
-  | { type: 'FETCH_INIT'; payload: string }
-  | { type: 'FETCH_SUCCESS'; payload: CommentDelete }
-  | { type: 'FETCH_FAILURE'; payload: string };
-
-const dataFetchReducer = (state: CommentDeleteState, action: Action): CommentDeleteState => {
-  switch (action.type) {
-    case 'FETCH_INIT':
-      return {
-        ...state,
-        isError: false,
-      };
-
-    case 'FETCH_SUCCESS':
-      return {
-        ...state,
-        isError: false,
-      };
-    case 'FETCH_FAILURE':
-      return {
-        ...state,
-        isError: true,
-      };
-    default:
-      return state;
-  }
-};
-
-interface DeleteComment {
+interface MutationArgs {
   caseId: string;
   commentId: string;
 }
 
-export interface UseDeleteComment extends CommentDeleteState {
-  deleteComment: ({ caseId, commentId }: DeleteComment) => void;
-}
-
-export const useDeleteComment = (): UseDeleteComment => {
-  const [state, dispatch] = useReducer(dataFetchReducer, {
-    isError: false,
-  });
+export const useDeleteComment = () => {
   const toasts = useToasts();
-  const isCancelledRef = useRef(false);
-  const abortCtrlRef = useRef(new AbortController());
   const refreshCaseViewPage = useRefreshCaseViewPage();
 
-  const dispatchDeleteComment = useCallback(
-    async ({ caseId, commentId }: DeleteComment) => {
-      try {
-        isCancelledRef.current = false;
-        abortCtrlRef.current.abort();
-        abortCtrlRef.current = new AbortController();
-        dispatch({ type: 'FETCH_INIT', payload: commentId });
-
-        await deleteComment({
-          caseId,
-          commentId,
-          signal: abortCtrlRef.current.signal,
-        });
-
-        if (!isCancelledRef.current) {
-          refreshCaseViewPage();
-          dispatch({ type: 'FETCH_SUCCESS', payload: { commentId } });
-        }
-      } catch (error) {
-        if (!isCancelledRef.current) {
-          if (error.name !== 'AbortError') {
-            toasts.addError(
-              error.body && error.body.message ? new Error(error.body.message) : error,
-              { title: i18n.ERROR_TITLE }
-            );
-          }
-          dispatch({ type: 'FETCH_FAILURE', payload: commentId });
-        }
-      }
+  return useMutation(
+    ({ caseId, commentId }: MutationArgs) => {
+      const abortCtrlRef = new AbortController();
+      return deleteComment({ caseId, commentId, signal: abortCtrlRef.signal });
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    []
-  );
-
-  useEffect(
-    () => () => {
-      isCancelledRef.current = true;
-      abortCtrlRef.current.abort();
-    },
-    []
+    {
+      mutationKey: casesMutationsKeys.deleteComment,
+      onSuccess: () => {
+        refreshCaseViewPage();
+      },
+      onError: (error: ServerError) => {
```

Comment:
You can use 

```
const { showErrorToast, showSuccessToast } = useCasesToast();


showErrorToast(error, { title: i18n.ERROR_TITLE });
```

Check `x-pack/plugins/cases/public/containers/use_delete_cases.tsx` for example.